src/bvh/bvh_impl.rs

```diff
@@ -165,6 +166,32 @@ impl<T: BHValue, const D: usize> Bvh<T, D> {
         DistanceTraverseIterator::new(self, ray, shapes)
     }
 
+    /// Traverses the [`Bvh`].
+    /// Returns the shape the closest to the query point and the distance to it.
+    ///
+    ///
+    /// [`Bvh`]: struct.Bvh.html
+    /// [`Aabb`]: ../aabb/struct.Aabb.html
+    ///
+    pub fn nearest_from<'a, Shape: Bounded<T, D> + PointDistance<T, D>>(
+        &self,
+        origin: nalgebra::Point<T, D>,
+        shapes: &'a [Shape],
+    ) -> Option<(&'a Shape, T)>
+    where
+        Self: std::marker::Sized,
+    {
+        if shapes.is_empty() {
```

Comment:
To be safe, considering the caller can manipulate `shapes`, let's use `self.nodes.is_empty()` instead.